core/transaction.go

```diff
@@ -26,10 +31,40 @@ type DeployTransaction struct {
 	Version *felt.Felt
 }
 
-func (d *DeployTransaction) Hash() *felt.Felt {
-	// Todo: implement pedersen hash as defined here:
+func (d *DeployTransaction) Hash(chainId []byte) (*felt.Felt, error) {
```

Comment:
We can shorten this function by doing something like this:
```go
func (d *DeployTransaction) Hash(chainId *felt.Felt) (*felt.Felt, error) {
	snKeccakContructor, err := crypto.StarkNetKeccak([]byte("constructor"))
	if err != nil {
		return nil, err
	}
	return crypto.PedersenArray(
		new(felt.Felt).SetBytes([]byte("deploy")),
		d.Version,
		d.ContractAddress,
		snKeccakContructor,
		crypto.PedersenArray(d.ConstructorCalldata...),
		new(felt.Felt),
		chainId,
	), nil
}
```

Reply:
Done